### PR TITLE
*: add user guides for DM

### PR DESCRIPTION
**Pull request by IANTHEREAL**
### What problem does this PR solve? 
improve DM document and make it friendly to read and install DM, now documents has two problem
- document structure, let it fall into two broad categories
  - user guides
  - user manual
- lack of troubleshoot document to tell user how to solve problem in DM setup or task starting, how to read log and monitoring. I add these document, we can fill and improve them later

### What is changed and how it works?
add user guides
adjust structure of document
### Check List 

Tests 
 - No code

### CODE REVIEW

**Comment by IANTHEREAL** [comment]
@shenli @LeopPro @csuzhangxc PTAL, I don't fill some contents now, I would fill them later

File: docs/dm/zh_CN/README.md [hunk 1]
```diff
@@ -5,14 +5,25 @@ DM (Data Migration) 用户文档
...
+DM (Data Migration) 是基于 mydumper / loader / syncer 的调度管理一体化数据同步平台，设计的主要目的是
+   - 标准化 （e.g. 工具运行，错误定义）
+   - 降低运维使用成本
+   - 简化错误处理流程
+   - 提升产品使用体验
+
+### 用户指引
 
 - [Get Started](./get-started.md)
+- [syncer/loader 升级](./upgrade-to-dm)
```
Comment (shenli) [inline docs/dm/zh_CN/README.md]: 从 syncer/loader 升级

File: docs/dm/zh_CN/configuration/configuration.md [hunk 2]
```diff
@@ -122,7 +122,7 @@ mysql-instances:
     ​    instance-id: "instance118-5306"
 
     ​    mydumper:                                                          # mydumper 相关配置（不能和 mydumper-config-name 同时设置）
-    ​        mydumper-path: "./mydumper"                                    # 见上文 mydumpers 配置项说明
+    ​        mydumper-path: "./mydumper"                                    # mydumper binary 文件地址，这个无需设置，会由 ansible 部署程序自动生成
```
Comment (shenli) [inline docs/dm/zh_CN/configuration/configuration.md:125]: 那么删掉这项配置？

File: docs/dm/zh_CN/get-started.md [hunk 3]
```diff
@@ -1,22 +1,56 @@
 Get Started
 ===
 
-### 1. 部署 DM 集群
+### 内容索引
+- [介绍 DM](#介绍-dm)
```
Comment (shenli) [inline docs/dm/zh_CN/get-started.md]: 介绍 DM -> 简介

File: docs/dm/zh_CN/configuration/configuration.md (hunk 2 above)
Comment (IANTHEREAL) [reply docs/dm/zh_CN/configuration/configuration.md:125]: 从 guide 里面删除了，留在 configuration 细节解释的原因是，怕客户手动部署

File: docs/dm/zh_CN/get-started.md [hunk 4]
```diff
@@ -135,30 +169,30 @@ start-task task.yaml            # task.yaml 为上一步编辑的配置文件路
...
 ```bash
 stop-task test              # 其中的 `test` 是 `task.yaml` 配置文件中 `name` 配置项设置的任务名
 ```
 
 
-### 3. 监控与 log
+### 监控与日志
 
-#### 3.1 DM 监控 dashboard
+#### 3DM 监控 dashboard
```
Comment (amyangfei) [inline docs/dm/zh_CN/get-started.md]: s/3DM/1 DM

File: docs/dm/zh_CN/get-started.md [hunk 5]
```diff
@@ -135,30 +169,30 @@ start-task task.yaml            # task.yaml 为上一步编辑的配置文件路
...
 
 假设参考 [1. 部署 DM 集群](#1-部署-dm-集群) 部署 DM 集群时，正确部署了 prometheus 与 grafana，且 grafana 的地址为 `172.16.10.71`。
 
 在浏览器中打开 <http://172.16.10.71:3000> 进入 grafana，选择 DM 的 dashboard 即可看到 DM 相关监控项，具体各监控项的解释参见 [DM 监控与告警](./maintenance/metrics-alert.md)。
 
-#### 3.2 DM log
+#### DM log
```
Comment (amyangfei) [inline docs/dm/zh_CN/get-started.md:195]: ditto s/DM log/2 DM log

File: docs/dm/zh_CN/troubleshoot/README.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+DM (Data Migration) 问题处理手册
+===
+
+如果你在运行 DM 的过程中遇到了错误，你可以尝试做下面的几件事情
+
+1. 查看日志文件，检查与你遇到的错误相关的日志内容。日志文件分布在 dm-master、dm-worker 部署节点上。 然后可以查看 common errors 寻找相应的解决方案。
```
Comment (amyangfei) [inline docs/dm/zh_CN/troubleshoot/README.md]: add a link to `common errors`?

File: docs/dm/zh_CN/troubleshoot/README.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+2. 如果没有找到类似的常见错误解决方案，并且不能通过日志或者监控自己解决问题，那么你可以联系对应的销售支持人员。
+
+3. 当你处理完错误后，一般情况下只要利用 `dmctl` 重启任务即可
+```bash
+resume-task ${task name}
+```
+   但是在一些情况下，你不得不重置数据同步任务，可以按照[重制任务](./reset-task.md)操作
```
Comment (amyangfei) [inline docs/dm/zh_CN/troubleshoot/README.md]: s/重制/重置

File: docs/dm/zh_CN/get-started.md [hunk 8]
```diff
@@ -135,30 +169,30 @@ start-task task.yaml            # task.yaml 为上一步编辑的配置文件路
...
 ```
 
 
-### 3. 监控与 log
+### 监控与日志
 
-#### 3.1 DM 监控 dashboard
+#### 3DM 监控 dashboard
 
 假设参考 [1. 部署 DM 集群](#1-部署-dm-集群) 部署 DM 集群时，正确部署了 prometheus 与 grafana，且 grafana 的地址为 `172.16.10.71`。
```
Comment (amyangfei) [inline docs/dm/zh_CN/get-started.md]: `[1. 部署 DM 集群](#1-部署-dm-集群) ` link has changed to `./get-started.md#部署-dm-集群`

File: docs/dm/zh_CN/README.md [hunk 9]
```diff
@@ -32,9 +43,9 @@ DM (Data Migration) 用户文档
     - [分库分表支持方案](./shard-table/merge-solution.md)
     - [sharding DDL 使用限制](./shard-table/restrictions.md)
     - [手动处理 sharding DDL lock](./shard-table/handle-DDL-lock.md)
-- [错误处理](./exception-handling)
-    - [skip 或 replace 异常 SQL](./exception-handling/skip-replace-sqls.md)
-    - [重置数据同步任务](./exception-handling/reset-task.md)
-- [从老版本工具升级到 DM](./upgrade-to-dm)
-    - [loader 升级到 DM](./upgrade-to-dm/upgrade-from-loader.md)
-    - [syncer 升级到 DM](./upgrade-to-dm/upgrade-from-syncer.md)
+- [错误处理](./troubleshoot)
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/README.md:46]: some links in FAQ need to update

File: docs/dm/zh_CN/maintenance/metrics-alert.md [hunk 10]
```diff
@@ -3,6 +3,14 @@ DM 监控与告警
 
 目前只有 dm-worker 提供了 metrics， dm-master 暂未提供。
 
+### 内容索引
+- [任务监控项](#Task)
+- [Relay Log Unit 监控项](#Relay-Log-Unit)
+- [Mydumper Unit 监控项](#Mydumper-Unit)
+- [Load Unit 监控项](#Load-Unit)
+- [Binlog-Replication-Unit 监控项](#Binlog-Replication-Unit)
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/maintenance/metrics-alert.md]: should we need to unify the name? (sync unit or binlog replication unit)

File: docs/dm/zh_CN/maintenance/metrics-alert.md [hunk 11]
```diff
@@ -64,11 +50,21 @@ query: `changes(dm_relay_write_error_count{instance="$instance"}[1m])`
 
 告警：需要告警
 
-#### binlog file index
+#### storage remain
 
-query: `dm_relay_binlog_file{instance="$instance"}`
+query: `dm_relay_space{instance="$instance", type="available"}`
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/maintenance/metrics-alert.md:55]: what is the logic for metrics' order? should we need to update the grafana dashboard?

File: docs/dm/zh_CN/troubleshoot/common-errors.md [hunk 12]
```diff
@@ -0,0 +1,11 @@
+常见错误解决方案
+===
+
+1. query-task 或者查看 log 发现 `Access denied for user 'root'@'172.31.43.27' (using password: YES)`
+
+   在 DM 的所有相关配置文件中，数据库相关的密码需要使用 dmctl 加密后的密文（如果数据库密码为空，则不需要加密）。 
+
+   了解如何使用 dmctl 加密明文密码，可以参考 [dmctl 加密上游 MySQL 用户密码](./maintenance/dm-ansible.md#dmctl-加密上游-mysql-用户密码)。
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/troubleshoot/common-errors.md]: the path for `dm-ansible.md` is `../maintenance/`, not `./maintenance/`

File: docs/dm/zh_CN/troubleshoot/common-errors.md [hunk 13]
```diff
@@ -0,0 +1,11 @@
...
+
+   此外，DM 在运行过程中，相关的上下游数据库用户需要具备相应的读写权限。DM 在启动任务过程中，也会自动进行部分权限检查，具体见 [上游 MySQL 实例权限](./task-handling/check-mysql.md)。
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/troubleshoot/common-errors.md]: `../` too

File: docs/dm/zh_CN/maintenance/metrics-alert.md (hunk 11 above)
Comment (IANTHEREAL) [reply docs/dm/zh_CN/maintenance/metrics-alert.md:55]: It's description docuement, put critical metrics in front, not necessarily a one-to-one correspondence with grafana, but I want to fill these description into grafana dashboard long time, file a issue for it.

File: docs/dm/zh_CN/maintenance/metrics-alert.md (hunk 10 above)
Comment (IANTHEREAL) [reply docs/dm/zh_CN/maintenance/metrics-alert.md]: it should be binlog replication unit, sync is not a good name

File: docs/dm/zh_CN/maintenance/metrics-alert.md [hunk 14]
```diff
@@ -3,6 +3,14 @@ DM 监控与告警
 
 目前只有 dm-worker 提供了 metrics， dm-master 暂未提供。
 
+### 内容索引
+- [任务监控项](#Task)
+- [Relay Log Unit 监控项](#Relay-Log-Unit)
+- [Mydumper Unit 监控项](#Mydumper-Unit)
+- [Load Unit 监控项](#Load-Unit)
+- [Sync/Binlog Replication Unit 监控项](#Binlog-Replication-Unit)
```
Comment (amyangfei) [inline docs/dm/zh_CN/maintenance/metrics-alert.md:11]: remove `Sync/` from `Sync/Binlog Replication Unit` ?

**Review by amyangfei — COMMENTED** [review]
Rest LGTM

File: docs/dm/zh_CN/get-started.md [hunk 15]
```diff
@@ -73,8 +108,8 @@ mysql-instances:                                  # 当前任务需要使用的
     port: 3306
     user: "root"
     password: "VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU="  # 明文 `123456` 某次加密后的密文，每次加密产生的密文会不同
-  instance-id: "172.16.10.81:3306"                # MySQL-1 的实例 ID，与 dm-master.toml 中的 `mysql-instance` 对应
-  black-white-list: "global"                      # 需要同步的库名/表名黑白名单的配置项名称，用于引用全局的黑白名单配置
+  instance-id: "172.16.10.81:3306"                # MySQL-1 的实例 ID，必須与 dm-master.toml 中的 `mysql-instance` 对应
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/get-started.md]: `必須`，Traditional Chinese?

**Comment by csuzhangxc** [comment]
rest LGTM

File: docs/dm/zh_CN/maintenance/metrics-alert.md (hunk 14 above)
Comment (IANTHEREAL) [reply docs/dm/zh_CN/maintenance/metrics-alert.md:11]: now we don't replace sync by binlog replication unit completely, leave it and replace it in anywhere later

**Comment by IANTHEREAL** [comment]
@amyangfei @csuzhangxc @shenli PTAL again

**Comment by csuzhangxc** [comment]
LGTM

**Review by amyangfei — COMMENTED** [review]
LGTM

**Pull request merged**